src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -543,6 +555,28 @@ private void UpdateSortIndex(FhirCosmosResourceWrapper cosmosWrapper)
             }
         }
 
+        private static string RemoveTrailingZerosFromMilisecondsForAGivenDate(DateTimeOffset date)
+        {
+            // 0000000+ -> +, 0010000+ -> 001+, 0100000+ -> 01+, 0180000+ -> 018+, 1000000 -> 1+, 1100000+ -> 11+, 1010000+ -> 101+
+            // ToString("o") - Formats to 2022-03-09T01:40:52.0690000+02:00 but serialized value to string in dB is 2022-03-09T01:40:52.069+02:00
+            var formattedDate = date.ToString("o", CultureInfo.InvariantCulture);
+            var milliseconds = formattedDate.Substring(20, 7); // get 0690000
+            var trimmedMilliseconds = milliseconds.TrimEnd('0'); // get 069
+            if (milliseconds.Equals("0000000", StringComparison.Ordinal))
+            {
+                // when date = 2022-03-09T01:40:52.0000000+02:00, value in dB is 2022-03-09T01:40:52+02:00, we need to replace the . after second
+                return formattedDate.Replace("." + milliseconds, trimmedMilliseconds, StringComparison.Ordinal);
```

Comment:
Instead of trimmedMilliseconds, could we explicitly use the empty string? I believe if we hit this return statement trimmedMilliseconds will always be the empty string.